src/types/item_id.rs

```diff
@@ -0,0 +1,112 @@
+use super::TweakDbId;
+use crate::raw::root::RED4ext as red;
+
+const DEFAULT_ITEM_ID_RNG_SEED: u32 = 2;
+
+#[derive(Default, Clone, Copy)]
+#[repr(transparent)]
+pub struct ItemId(red::ItemID);
+
+impl ItemId {
+    #[inline]
+    pub const fn new_from(id: TweakDbId) -> Self {
+        Self(red::ItemID {
+            tdbid: id.to_inner(),
+            rngSeed: DEFAULT_ITEM_ID_RNG_SEED,
+            uniqueCounter: 0,
+            structure: GamedataItemStructure::BlueprintStackable as u8,
+            flags: GameEItemIdFlag::None as u8,
+        })
+    }
+
+    pub fn structure(&self) -> GamedataItemStructure {
+        self.0.structure.into()
+    }
+
+    pub fn flags(&self) -> GameEItemIdFlag {
+        self.0.flags.into()
+    }
+
+    #[inline]
+    pub fn get_tdbid(&self) -> TweakDbId {
```

Comment:
I'd rename it for consistency, the other methods are also getters but aren't prefixed with `get_`
```suggestion
    pub fn tdbid(&self) -> TweakDbId {
```

Reply:
renamed in 413fee2